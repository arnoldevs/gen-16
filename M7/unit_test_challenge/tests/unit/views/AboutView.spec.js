import { mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import AboutView from "@/views/AboutView.vue";

describe("AboutView", () => {
	it("renders a heading", async () => {
		const router = createRouter({
			history: createWebHistory(),
			routes: [
				{
					path: "/about",
					name: "about",
					component: AboutView,
				},
			],
		});

		router.push("/about");
		await router.isReady();

		const wrapper = mount(AboutView, {
			global: {
				plugins: [router],
			},
		});

		expect(wrapper.find("h1").text()).toBe("Acerca de");
	});
});

describe("AboutView", () => {
	it("matches snapshot", () => {
		const wrapper = mount(AboutView);
		expect(wrapper.html()).toMatchSnapshot();
	});
});
