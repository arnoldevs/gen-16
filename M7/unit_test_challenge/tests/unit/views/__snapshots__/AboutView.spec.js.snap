// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView matches snapshot 1`] = `
<div class="about">
  <h1>Acerca de</h1>
</div>
`;
